### PR TITLE
calc timesheet dates on validate not just submit

### PR DESCRIPTION
**Pull request by RobertSchouten**

### CODE REVIEW

**Comment by nabinhait** [comment]
@rohitwaghchaure Can you review this PR?

File: erpnext/projects/doctype/timesheet/timesheet.py [hunk 1]
```diff
@@ -28,6 +28,7 @@ def validate(self):
 		self.update_cost()
 		self.calculate_total_amounts()
 		self.calculate_percentage_billed()
+		self.set_dates()
```
Comment (rohitwaghchaure) [inline erpnext/projects/doctype/timesheet/timesheet.py]: Adding function under validate will execute on every save. Start date and end date is used in salary slip while making it from timesheet

Comment (RobertSchouten) [inline erpnext/projects/doctype/timesheet/timesheet.py]: the idea behind this is to keep a consitancy seems wierd to not have on save only submit and i wanted to use for a customisation but i could easily add another way

**Comment by rohitwaghchaure** [comment]
@nabinhait 
Reviewed, please merge

**Review by nabinhait — APPROVED** [review]

**Pull request merged**